app/frontend/src/service/user.ts

```diff
@@ -20,11 +20,30 @@ import { ProtoToJsTypes } from "../utils/types";
 import client from "./client";
 
 type RequiredUpdateProfileReq = Required<UpdateProfileReq.AsObject>;
-export type ProfileFormData = {
+type ProfileFormData = {
   [K in keyof RequiredUpdateProfileReq]: ProtoToJsTypes<
     RequiredUpdateProfileReq[K]
   >;
 };
+
+export type UpdateUserProfileData = Pick<
+  ProfileFormData,
+  | "name"
+  | "city"
+  | "gender"
+  | "occupation"
+  | "languages"
+  | "aboutMe"
+  | "aboutPlace"
+  | "countriesVisited"
+  | "countriesLived"
+>;
+
+export type HostingPreferenceData = Omit<
+  ProfileFormData,
+  keyof UpdateUserProfileData | "color" | "hostingStatus"
+>;
```

Comment:
Made these new two types to make testing easier without type asserting too much, otherwise I'd have had to set a lot of unrelated fields. It's probably more accurate types for the respective form code anyway.